server/plugin_test.go

```diff
@@ -38,8 +38,17 @@ func TestPlugin(t *testing.T) {
 func TestSpecialCases(t *testing.T) {
 	links := make([]*Link, 0)
 	links = append(links, &Link{
+		Pattern:  "https://mattermost.com",
+		Template: "[the mattermost portal](https://mattermost.com)",
```

Comment:
What is "the mattermost portal? Why do we need to auto-link this?

Reply:
This is just a test example. Basically you can have this autolinked, but you can have this "auto-linked with a specific text" instead of the url.